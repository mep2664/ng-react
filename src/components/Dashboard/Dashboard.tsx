import * as React from "react";
import { Link } from "react-router-dom";

export const Dashboard: React.FC = () => {
    return (
        <div>
            dashboard
            <Link to={{ pathname: "/ticket", search: "?project=1&ticket=1" }}>NEWGIRRA-1</Link>
        </div>
    );
}