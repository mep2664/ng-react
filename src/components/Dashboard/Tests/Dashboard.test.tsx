import React from "react"
import { render, wait } from "@testing-library/react"
import "@testing-library/jest-dom/extend-expect";
import { Dashboard, GET_DATA } from "../Dashboard"
import { MockedProvider } from '@apollo/react-testing';

const tickets = [
    {
        projectName: "PROJECT_NAME", ticketNumber: "1", ticketId: "TICKET_ID_1",
        description: "DESCRIPTION", sprintName: "SPRINT_NAME", priority: "PRIORITY",
        ticketType: "TICKET_TYPE", storyPoints: "STORY_POINTS"
    },
    {
        projectName: "PROJECT_NAME", ticketNumber: "2", ticketId: "TICKET_ID_2",
        description: "DESCRIPTION", sprintName: "SPRINT_NAME", priority: "PRIORITY",
        ticketType: "TICKET_TYPE", storyPoints: "STORY_POINTS"
    },
];

const mocks = [
    {
        request: {
            query: GET_DATA,
            variables: {},
        },
        result: {
            data: {
                allTickets: tickets,
            },
        },
    },
];

jest.mock("../../TicketDetail", () => {
    return {
        TicketDetail: () => <span>TICKET_DETAIL</span>,
    }
});

jest.mock("../../Loaders/Loader", () => {
    return {
        Loader: () => <span>loading...</span>,
    }
});

describe("Tests for the Dashboard component", () => {
    test("Component matches snapshot", async () => {
        const { container } = render(
            <MockedProvider mocks={mocks} addTypename={false}>
                <Dashboard />
            </MockedProvider>,
        );
        await wait();
        expect(container).toMatchSnapshot();
    });

    test("Displays loading when waiting for data", () => {
        const { getByText } = render(
            <MockedProvider mocks={mocks} addTypename={false}>
                <Dashboard />
            </MockedProvider>);
        expect(getByText("loading...")).toBeDefined();
    });

    test("Displays loading when waiting for data", async () => {
        const errorMock = {
            request: {
                query: GET_DATA,
                variables: {},
            },
            error: new Error("something went wrong..."),
        };
        const { getByText } = render(
            <MockedProvider mocks={[errorMock]} addTypename={false}>
                <Dashboard />
            </MockedProvider>);
        await wait()
        expect(getByText("Network error: something went wrong...")).toBeDefined();
    });

    test("Component renders TicketDetails", async () => {
        const { getAllByText } = render(
            <MockedProvider mocks={mocks} addTypename={false}>
                <Dashboard />
            </MockedProvider>,
        );
        await wait();
        expect(getAllByText("TICKET_DETAIL").length).toBe(2);
    });
});
