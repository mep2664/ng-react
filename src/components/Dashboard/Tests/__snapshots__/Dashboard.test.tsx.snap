// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests for the Dashboard component Component matches snapshot 1`] = `
<div>
  <div>
    <div
      class="sc-Rmtcm jNKcBE"
    >
      <span>
        TICKET_DETAIL
      </span>
      <span>
        TICKET_DETAIL
      </span>
    </div>
  </div>
</div>
`;
